.github/workflows/publish.yml

```diff
@@ -0,0 +1,39 @@
+name: Publish package to the Maven Central Repository
+on:
+  workflow_dispatch:
+    inputs:
+      version:
+        description: 'Version to publish. For example "0.0.1"'
+        required: true
+        default: '0.0.0'
+  push:
+    tags:
+      - v*
```

Comment:
Creating a tag will trigger this publishing, I think what you actually want is to trigger when a new GH release is created or, even more strict, published. ([reference](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#release))

Reply:
Awesome, will give it a go.